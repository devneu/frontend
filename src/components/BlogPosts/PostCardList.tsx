import React from 'react';
import { Col, Row } from 'antd';
import SmallPostCard from './SmallPostCard';

interface PostCardListProps {
  postInfo: {
    id: number;
    postImg: string;
    postTitle: string;
    postText: string;
    postDate: string;
    sizeBlock: number;
  };
}

const PostCardList = ({ postInfo }: PostCardListProps) => (
  <Row justify="space-around" align="middle">
    <Col span={14}>
      <Row justify="space-between">
        {
          // @ts-ignore
          // eslint-disable-next-line react/destructuring-assignment,max-len
          postInfo.map(
            (
              obj: {
                id?: number;
                postImg?: string;
                postDate?: string;
                postTitle?: string;
                postText?: string;
              },
              index: number,
            ) => (
              <Col span={index === 0 ? 24 : 11}>
                <SmallPostCard key={obj.id} post={obj} />
              </Col>
            ),
          )
        }
      </Row>
    </Col>
  </Row>
);

export default PostCardList;
