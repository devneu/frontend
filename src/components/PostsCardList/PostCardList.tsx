import React from 'react';
import { Col, Row } from 'antd';
import SmallPostCard from '../Posts/SmallPostCard';
import { Post } from '../../types/Post.interface';
import FeaturePost from '../Posts/FeaturePost';

interface PostsCardListProps {
  posts: Post[];
  featurePost: Post;
}

const PostsCardList = ({ posts, featurePost }: PostsCardListProps) => (
  <Row gutter={[24, 24]}>
    <Col span={24}>
      <FeaturePost post={featurePost} />
    </Col>
    {posts.map((post) => (
      <Col span={12} key={post.id}>
        <SmallPostCard post={post} />
      </Col>
    ))}
  </Row>
);

export default PostsCardList;
