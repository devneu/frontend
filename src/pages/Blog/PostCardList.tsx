import {Col, Row} from "antd";
import {postInfo} from "./data";
import SmallPostCard from "./SmallPostCard";
import React from "react";

const PostCardList  = () => {
    return (
        <Row justify="space-around" align="middle">
            <Col span={14}>
                <Row justify="space-between">
                    {
                        postInfo.map((obj, index) => <Col span={index==0 ? 24: 11}><SmallPostCard key={obj.id} post={obj}/></Col>)
                    }
                </Row>
            </Col>
            <Col span={6}></Col>
        </Row>
    );
}

export default PostCardList;