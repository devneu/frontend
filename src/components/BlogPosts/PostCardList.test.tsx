import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import PostCardList from './PostCardList';

describe('PostCardList', () => {
  it('Should successfully render', () => {
    const wrapper = renderer
      .create(
        <BrowserRouter>
          <PostCardList
            posts={[
              {
                id: 1,
                postImg: 'post image',
                postDate: ' post date',
                postTitle: ' post title',
                postText: 'post text',
              },
            ]}
            featurePost={{
              id: 1,
              postImg: 'post image',
              postDate: ' post date',
              postTitle: ' post title',
              postText: 'post text',
            }}
          />
        </BrowserRouter>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
