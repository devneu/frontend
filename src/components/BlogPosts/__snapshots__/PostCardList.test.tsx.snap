// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCardList Should successfully render 1`] = `
<div
  className="ant-row"
  style={
    Object {
      "marginBottom": -12,
      "marginLeft": -12,
      "marginRight": -12,
      "marginTop": -12,
    }
  }
>
  <div
    className="ant-col ant-col-24"
    style={
      Object {
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      }
    }
  >
    <div
      className="ant-card ant-card-bordered post-feature-card"
    >
      <div
        className="ant-card-cover"
      >
        <img
          alt="card-img"
          src="post image"
        />
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-meta"
        >
          <div
            className="ant-card-meta-detail"
          >
            <div
              className="ant-card-meta-description"
            >
               post date
            </div>
          </div>
        </div>
        <h2
          className="ant-typography"
          style={
            Object {
              "WebkitLineClamp": undefined,
            }
          }
        >
           post title
        </h2>
        <div
          className="ant-typography post-text"
          style={
            Object {
              "WebkitLineClamp": undefined,
            }
          }
        >
          post text
        </div>
        <a
          href="/blog/1"
          onClick={[Function]}
        >
          Read more
        </a>
      </div>
    </div>
  </div>
  <div
    className="ant-col ant-col-12"
    style={
      Object {
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      }
    }
  >
    <div
      className="ant-card ant-card-bordered post-card"
    >
      <div
        className="ant-card-cover"
      >
        <img
          alt="card-img"
          src="post image"
        />
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-meta"
        >
          <div
            className="ant-card-meta-detail"
          >
            <div
              className="ant-card-meta-description"
            >
               post date
            </div>
          </div>
        </div>
        <h2
          className="ant-typography"
          style={
            Object {
              "WebkitLineClamp": undefined,
            }
          }
        >
           post title
        </h2>
        <div
          className="ant-typography post-text"
          style={
            Object {
              "WebkitLineClamp": undefined,
            }
          }
        >
          post text
        </div>
        <a
          href="/blog/1"
          onClick={[Function]}
        >
          Read more
        </a>
      </div>
    </div>
  </div>
</div>
`;
